src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -2817,6 +2818,19 @@ private void ImportBinaryOperation(ILOpcode opcode)
             PushExpression(kind, "binop", result, type);
         }
 
+        private TypeDesc WidenBytesAndShorts(TypeDesc type)
+        {
+            if (type == GetWellKnownType(WellKnownType.Byte)
```

Comment:
It would be more efficient to do this as a switch on `type.Category`

Reply:
Thanks, have changed to a `switch`.